internal/api/cluster_installation.go

```diff
@@ -122,6 +141,68 @@ func handleSetClusterInstallationConfig(c *Context, w http.ResponseWriter, r *ht
 		return
 	}
 
-	// TODO: write the config
-	w.WriteHeader(http.StatusNotImplemented)
+	cluster, err := c.Store.GetCluster(clusterInstallation.ClusterID)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to query cluster")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if cluster == nil {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	var applyConfig func(parentKey string, value map[string]interface{}) error
+
+	// applyConfig takes the decomposed configuration, walks the resulting map, and invokes
+	// something akin to:
+	//	mattermost config set <ParentKey1.ParentKey2.LeafKey> <value>
+	//
+	// Ideally, this would be replaced by simply using the API and passing in the config struct
+	// directly, but at the moment that requires authentication.
+	applyConfig = func(parentKey string, parentValue map[string]interface{}) error {
```

Comment:
A little annoying that we have to do this, I thought we had a CLI for setting the entire config but it doesn't seem we do. Maybe a future improvement to the CLI.